frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/VerifyCodeHandler.java

```diff
@@ -197,6 +197,11 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
                         errorResponse.get(),
                         session,
                         auditContext);
+
+                if (journeyType == JourneyType.REAUTHENTICATION
+                        && checkErrorCountsForReauth(journeyType, userProfileMaybe)) {
+                    return generateApiGatewayProxyErrorResponse(400, ErrorResponse.ERROR_1057);
```

Comment:
A note that we should also emit the reauth failed audit event here (and in the other handler). I think it might be easier to come back to this once Aidan's PRs are in

Reply:
Also same comment as on the VerifyMfaCodeHandler - would be nice to see if we can get this logic in the same place as the other error codes, while acknowledging that in general we'd like to do a refactor to get this stuff done in less surprising places!

Reply:
Let's defer anymore change on this until the refactor.